harness/helpers.ts

```diff
@@ -277,3 +239,49 @@ export class TestClient {
         })
     }
 }
+
+export class CloudTestClient extends BaseTestClient {
+    async createClient(options: Record<string, unknown> = {}, sdkKey?: string | null) {
+        if (sdkKey !== undefined) {
+            this.sdkKey = sdkKey
+        }
+        const response = await createClient(
+            getConnectionStringForProxy(this.sdkName),
+            this.clientId,
+            this.sdkKey,
+            {
+                baseURLOverride: `${getMockServerUrl()}/client/${this.clientId}`,
+                enableCloudBucketing: true,
+                ...options
+            }
+        )
+
+        this.clientLocation = response.headers.get('location')
```

Comment:
do we still need to respond with the location header from the `/client` endpoint? The test harness is the one that chooses the clientId so it should just know the location i.e. `/client/${clientId}`

Reply:
The test harness knows the clientId but it shouldn't assume it knows the location. Some of the other code here is assuming that the client path is `/client/${clientId}` when it should be respecting the location header from the proxy

Reply:
That's a valid point. But also some of the tests kind of need to assume what the location is for the mocked endpoints - the config endpoints needs to be mocked before the client is created otherwise the sdk is gonna receive a 404 for its first fetch. It's a little weird but I agree that the harness shouldn't assume the location

Reply:
the config endpoint is the concern of the harness and isn't really related to the "location" of the client on the proxy. That location is owned by the proxy